python/ray/serve/http_proxy.py

```diff
@@ -79,12 +79,13 @@ async def _send_request_to_handle(handle, scope, receive, send):
                          f"{MAX_REPLICA_FAILURE_RETRIES} retries.")
         await Response(
             error_message, status_code=500).send(scope, receive, send)
-        return
+        return "500"
 
     if isinstance(result, starlette.responses.Response):
         await result(scope, receive, send)
     else:
         await Response(result).send(scope, receive, send)
+    return None
```

Comment:
```suggestion
    if isinstance(result, starlette.responses.Response):
        await result(scope, receive, send)
        return str(result.status_code)
    else:
        await Response(result).send(scope, receive, send)
        return "200"
```

Reply:
@simon-mo I think this new metrics should differ from `serve_deployment_error_counter` in each replica? That means the counter only count the error outside of deployment.

Reply:
Maybe you're right, but there're somewhere else to modify, I commit again.

Reply:
ah you are right.. we need to distinguish application error and user specified error. i don't think we have a formal definition yet. this diff (and your diff) at least helps giving find-grained error code.